axol/axol.py

```diff
@@ -714,46 +727,112 @@ def femka(message):
-        player = findplayer(message.from_user)
+
+# todo: replace with neovip
+love_chats = [debug_chat_id]
+
+
+@bot.message_handler(commands=["love_butts"])
+def love_buttons(message):
+    if message.from_user.id != message.chat.id:
+        bot.send_message(message.chat.id, "ДАВАЙ ПООБЩАЕМСЯ В ЛИЧКЕ ;)")
+        return
+    markup = teletypes.InlineKeyboardMarkup(row_width=2)
+    reg_button = teletypes.InlineKeyboardButton("💫 ЗАРЕГАТЬСЯ", callback_data="reg_data")
+    send_card_button = teletypes.InlineKeyboardButton("💌 ОТПРАВИТЬ ВАЛЕНТИНКУ", callback_data="card_data")
+    check_task_button = teletypes.InlineKeyboardButton("💘 УЗНАТЬ ЗАДАНИЕ", callback_data="task_data")
+    markup.add(*[reg_button, send_card_button, check_task_button])
+
+    bot.send_message(message.chat.id, "ЧЕГО ТЕБЕ, КОТИК?", reply_markup=markup)
+
+
+@bot.callback_query_handler(func=lambda call: call.data == "reg_data")
+def reg_callback(call):
+    love_reg(call.message, call.from_user)
+
+
+def love_reg(message, user):
+    player = findplayer(user)
+    if not player.islove:
         player.islove = True
         bot.send_message(message.chat.id, "СПАСИБО ЗА РЕГИСТРАЦИЮ, КОТИК \u2764 \u2764 \u2764")
+    else:
+        bot.send_message(message.chat.id, "ТЫ УЖЕ В СПИСКЕ, ОЖИДАЙ НАЧАЛА ИГРЫ \u2764")
 
 
-@bot.message_handler(commands=["love_send"])
-def love_send(message):
-    text = str(message.text[11:])
-    if not message.from_user.id == message.chat.id:
-        return
-    if not text or text == "rakon_bot":
-        bot.send_message(message.chat.id, "НЕ СТЕСНЯЙСЯ, ВЫРАЗИ СВОИ ЧУВСТВА!")
-        return
-    text = "#валентинка" + "\n" + text
-    try:
-        bot.send_message(vip_chat_id, text)
-    except telebot.apihelper.ApiException:
-        bot.send_message(message.chat.id, "НЕ ВЫШЛО ОТПРАВИТЬ СООБЩЕНИЕ :(")
+@bot.callback_query_handler(func=lambda call: call.data == "card_data")
+def card_callback(call):
+    force_send_card = teletypes.ForceReply()
+    bot.send_message(call.message.chat.id, "С УДОВОЛЬСТВИЕМ ДОСТАВЛЮ ТВОЮ ВАЛЕНТИКУ! \u2764 "
+                                           "\nНАПИШИ ТЕКСТ ИЛИ ОТПРАВЬ <s>NUDES</s> ФОТОЧКУ, ГОЛОС, КРУГЛОВИДЕО: "
+                                           "Я ЗАБОТЛИВО УПАКУЮ И ДОСТАВЛЮ ЧТО УГОДНО. В ЧАТИК И АНОНИМНО ;) "
+                                           "\n\n ЕСЛИ НЕ ХОЧЕШЬ НИЧЕГО ОТПРАВЛЯТЬ, ПРОСТО НЕ ОТВЕЧАЙ НА ЭТО СООБЩЕНИЕ",
+                     reply_markup=force_send_card, parse_mode="HTML")
 
 
+@bot.message_handler(
+    content_types=['text', 'sticker', 'photo', 'video', 'video_note', 'voice', 'audio', 'document', 'animation'],
+    func=lambda message: check_valentine(message))
+def love_send(message):
+    hashtag = "#валентинка"
+    for chat in love_chats:
+        if message.content_type == 'text':
+            bot.send_message(chat, hashtag + "\n" + message.text)
+        else:
+            bot.send_message(chat, hashtag + "\n")
+            if message.content_type == 'sticker':
```

Comment:
I think this kind of rework relates more to refactoring, rather than to current issue. There are some similar places we can reuse, e. g. creation of a string of player's name, that is used not only in this game and just copy-pasted around. So I propose to think about it in another task (e. g. split_all, that is in progress).

Reply:
I agree that refactoring of the old code could be done outside of this PR. But this refactoring is related to new code. It's not about _using_ this function in other places now. Just simplifying love_send function.

Reply:
It's not quite the same sence that is in your first message. But your cursivechik makes me change my mind about this rework, so it'll be done.

Reply:
But not in this PR ))))))))